### PR TITLE
Calm down Eoin

### PR DESCRIPTION
**Pull request by joshuahigginson**

### CODE REVIEW

File: Doras/src/app/_services/authentication.service.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  logout() {
+    // remove user from local storage to log user out
+    localStorage.removeItem('currentUser');
+    this.currentUserSubject.next(null);
+    this.router.navigate(['/login']);
+  }
+
+  getToken() {
+    // tslint:disable-next-line:no-string-literal
```
Comment (e-mccoy) [inline Doras/src/app/_services/authentication.service.ts]: remove this

File: Doras/src/app/_services/authentication.service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    let r = new User();
```
Comment (e-mccoy) [inline Doras/src/app/_services/authentication.service.ts]: use a better var name than r

File: Doras/src/app/app.component.ts [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
-  constructor(private data: DataService) {}
+  currentUser: User;
+  constructor(
+    private authenticationService: AuthenticationService
+  ) {
+  }
```
Comment (e-mccoy) [inline Doras/src/app/app.component.ts]: move this lonely boi

File: Doras/src/app/app.module.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ import { LoginComponent } from './login/login.component';
 import { MainComponent } from './main/main.component';
 import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
 import { ForgotPasswordComponent } from './forgot-password/forgot-password.component';
+import { ErrorInterceptor} from './_helpers/error.interceptor';
```
Comment (e-mccoy) [inline Doras/src/app/app.module.ts:17]: space after `ErrorInterceptor`

File: Doras/src/app/login/login.component.ts [hunk 5]
```diff
@@ -12,24 +12,33 @@ import { Router } from '@angular/router';
 export class LoginComponent implements OnInit {
   newUser: User;
   checkUser: User;
-  constructor(private data: DataService, private router: Router) { }
+  constructor(
+    private data: DataService,
+    private authenticationService: AuthenticationService,
+  ) {
+  }
```
Comment (e-mccoy) [inline Doras/src/app/login/login.component.ts]: another lonely curly boi

File: ExpressAPI/app.js [hunk 6]
```diff
@@ -2,6 +2,66 @@ const express = require('express')
 const app = express()
 const env = require('dotenv').config()
 const db = require('./db.js')
+const bodyParser = require('body-parser');
+const b = require('bcryptjs');
```
Comment (e-mccoy) [inline ExpressAPI/app.js]: change this var name

File: ExpressAPI/server.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+const express = require('express');
```
Comment (e-mccoy) [inline ExpressAPI/server.js]: delet this

**Review by e-mccoy — APPROVED** [review]

**Review by e-mccoy — APPROVED** [review]

File: Doras/src/environments/environment.ts [hunk 8]
```diff
@@ -3,6 +3,7 @@
 // The list of file replacements can be found in `angular.json`.
 
 export const environment = {
+  apiUrl: 'http://localhost:4200',
```
Comment (micalexis) [inline Doras/src/environments/environment.ts]: this should be centralised to a config file so you dont have to change it in different places!

File: ExpressAPI/app.js [hunk 9]
```diff
@@ -2,6 +2,65 @@ const express = require('express')
...
+class HandlerGenerator {
+    login(req, res) {
+        let username = req.body.username;
+        let password = req.body.password;
+        db.getUser(username, function(rows) {
+            user = rows
+        })
+        if (username && password) {
+            if (username === user['username'] && bcrypt.compareSync(password, user['userPassword'])) {
+                let token = jwt.sign({
```
Comment (micalexis) [inline ExpressAPI/app.js]: const not let

**Review by micalexis — CHANGES_REQUESTED** [review]
minor changes required but looks good.
i would like a demo
squash commits

**Pull request merged**